File: random_algo_exercise/Armstrong.java

public class Armstrong
{

    // Calculate x raised to the power of y
int power(int x, long y)
{
  if (y==0)
    return 1;
  if(y%2 == 0)
    return power(x,y/2)* power(x,y/2);
  return x *power(x, y/2) * power(x, y/2);
  }

  //function to calculat order of the number

  int order(int x)
  {
    int n = 0;
    while (x != 0){
      n++;
      x= x/10;
    }
    return n;
  }
  //check whether the giveb number is Armstrong
  boolean isArmstrong (int x){
    int n = order(x);
    int temp = x,  sum = 0;

    while(temp != 0){
      int r = temp % 10;
      sum += power(r, n);
      temp = temp/10;

    }
    //if satisfies armstrong condition
    return (sum == x);
  }
  // Main program


public static void main(String[] args){

  Armstrong ob = new Armstrong();
  int x = 153;
  System.out.println(ob.isArmstrong(x));
}
}

File: chapter2_fundamentals_II/armstrong.py
n=int(input("Enter any number: "))
# a=list(map(int,str(n)))
# b=list(map(lambda x:x**3,a))
# if(sum(b)==n):
#     print("The number is an armstrong number. ")
# else:
#     print("The number isn't an arsmtrong number. ")

s =0
temp =n
while temp > 0:
    c = temp %10
    s += c**3
    temp //= 10
if n ==s:
    print("The number is an armstrong number. ")
else:
    print("The number isn't an arsmtrong number. ")
